src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -31,95 +30,62 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
+        /// Try to get the node that can be used to trigger the refactoring based on current cursor position. 
         /// </summary>
         /// <returns>
-        /// If the refactoring should be triggered, then returns the only namespace declaration node in the document (of type 
-        /// <typeparamref name="TNamespaceDeclarationSyntax"/>) or the compilation unit node (of type <typeparamref name="TCompilationUnitSyntax"/>)
-        /// if no namespace declaration in the document. Otherwise, return null.
+        /// (1) a node of type <typeparamref name="TNamespaceDeclarationSyntax"/> node, if curosr in the name and it's the 
+        /// only namespace declaration in the document.
+        /// (2) a node of type <typeparamref name="TCompilationUnitSyntax"/> node, if the cursor is in the name of first 
+        /// declaration in global namespace and there's no namespace declaration in this document.
+        /// (3) otherwise, null.
         /// </returns>
-        protected abstract Task<SyntaxNode> TryGetApplicableInvocationNode(Document document, int position, CancellationToken cancellationToken);
+        protected abstract Task<SyntaxNode> TryGetApplicableInvocationNodeAsync(Document document, TextSpan span, CancellationToken cancellationToken);
 
         protected abstract string EscapeIdentifier(string identifier);
 
-        protected abstract SyntaxList<TMemberDeclarationSyntax> GetMemberDeclarationsInContainer(SyntaxNode compilationUnitOrNamespaceDecl);
-
-        protected async Task<bool> ContainsPartialTypeWithMultipleDeclarationsAsync(
-            Document document, SyntaxNode compilationUnitOrNamespaceDecl, CancellationToken cancellationToken)
-        {
-            var memberDecls = GetMemberDeclarationsInContainer(compilationUnitOrNamespaceDecl);
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var semanticFacts = document.GetLanguageService<ISemanticFactsService>();
-
-            foreach (var memberDecl in memberDecls)
-            {
-                var memberSymbol = semanticModel.GetDeclaredSymbol(memberDecl, cancellationToken);
-
-                // Simplify the check by assuming no multiple partial declarations in one document
-                if (memberSymbol is ITypeSymbol typeSymbol
-                    && typeSymbol.DeclaringSyntaxReferences.Length > 1
-                    && semanticFacts.IsPartial(typeSymbol, cancellationToken))
-                {
-                    return true;
-                }
-            }
-            return false;
-        }
-
         private static string ChangeNamespaceActionTitle(State state)
             => state.TargetNamespace.Length == 0
                 ? FeaturesResources.Change_to_global_namespace
                 : string.Format(FeaturesResources.Change_namespace_to_0, state.TargetNamespace);
```

Comment:
is this called from multiple locations?  if not, just inline.

Reply:
Fixed